cmake/configure/configure_3_petsc.cmake

```diff
@@ -103,6 +103,20 @@ MACRO(FEATURE_PETSC_CONFIGURE_EXTERNAL)
   SET(DEAL_II_EXPAND_PETSC_BLOCKVECTOR "PETScWrappers::BlockVector")
   SET(DEAL_II_EXPAND_PETSC_MPI_VECTOR "PETScWrappers::MPI::Vector")
   SET(DEAL_II_EXPAND_PETSC_MPI_BLOCKVECTOR "PETScWrappers::MPI::BlockVector")
+  #
+  # FIXME:
+  # temporary variable until deal.II fully support complex-valued PETSc
+  # Note that petscconf.h should export PETSC_USE_COMPLEX 1 in complex-valued case.
+  FILE(READ ${PETSC_INCLUDE_DIRS}/petscconf.h _petscconf)
+  STRING(REGEX MATCH "\#define PETSC_USE_COMPLEX 1" _petsc_use_complex ${_petscconf})
+  IF( NOT _petsc_use_complex )
+    SET(DEAL_II_EXPAND_PETSC_VECTOR_REAL "PETScWrappers::Vector")
+    SET(DEAL_II_EXPAND_PETSC_BLOCKVECTOR_REAL "PETScWrappers::BlockVector")
+    SET(DEAL_II_EXPAND_PETSC_MPI_VECTOR_REAL "PETScWrappers::MPI::Vector")
+    SET(DEAL_II_EXPAND_PETSC_MPI_BLOCKVECTOR_REAL "PETScWrappers::MPI::BlockVector")
+  ELSE()
+    MESSAGE(STATUS "Compiling with complex-valued algebra")
+  ENDIF()
 ENDMACRO()
```

Comment:
damn it :smile: if you see here https://github.com/dealii/dealii/pull/2226#discussion_r53711475 (scroll down, it will be expanded), i was scratching my head why `PETSC_USE_COMPLEX` does not work (which is what the C++ define is), but it turns out that CMake decided to export `PETSC_WITH_COMPLEX`.